packages/process/test/daemon.test.ts

```diff
@@ -42,9 +41,14 @@ describe('daemon', () => {
     beforeEach(function*() {
       task.halt();
     });
-
     it('kills the process', function*() {
-      yield expect(fetch(`http://localhost:29000`, { method: "POST", body: "hello" })).rejects.toHaveProperty('name', 'FetchError');
+      let error;
+      try {
+        yield fetch(`http://localhost:29000`, { method: "POST", body: "hello" })
+      } catch (err) {
+        error = err;
+      }
+      expect(error).toHaveProperty('name', 'FetchError');
```

Comment:
It was returning the `Fetch` interface and not the response

Reply:
Ah, I see we're using the `Effection` fetch here. If we're going to treat the error as a value, then let's just capture it as a return value instead of re-assigning a variable.

```ts
let error = yield function*() {
  try {
    yield fetch(`http://localhost:29000`, { method: "POST", body: "hello" });
  } catch (error) {
    return error;
  }
}
expect(error).toMatchObject({ name: 'FetchError' });
```

This is more composable (decoupled from context) since it can be abstracted to handle any operation, not just a `fetch`. 

```ts
function* capture<T>(operation: Operation<T>): Operation<Error | T> {
  try {
    return yield operation;
  } catch (error) {
    return error;
  }
```

Which we can then use as 

```ts
expect(yield capture(fetch(`http://localhost:29000`, { method: "POST", body: "hello" }))).toMatchObject({ name: 'FetchError' });
```

In fact, I think there might be already a `captureError` operation in `@effection/mocha`

Reply:
👍 https://github.com/thefrontside/effection/pull/608/commits/cd7698e2dc3d8683fe55ce9f618777a9daa7f9b2